js/truncate_test.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { testPerm, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+function readDataSync(name: string): string {
+  const data = deno.readFileSync(name);
+  const decoder = new TextDecoder("utf-8");
+  const text = decoder.decode(data);
+  return text;
+}
+
+async function readData(name: string): Promise<string> {
+  const data = await deno.readFile(name);
+  const decoder = new TextDecoder("utf-8");
+  const text = decoder.decode(data);
+  return text;
+}
+
+testPerm({ write: true }, function truncateSyncSuccess() {
+  const enc = new TextEncoder();
+  const d = enc.encode("Hello");
+  const filename = deno.makeTempDirSync() + "/test_truncateSync.txt";
+  deno.writeFileSync(filename, d, 0o666);
+  
+  // test len more than file's length 
+  deno.truncateSync(filename, 20);
+  let data = readDataSync(filename);
+  assertEqual(data.length, 20);
+  // test len less than file's length
+  deno.truncateSync(filename, 5);
+  data = readDataSync(filename);
+  assertEqual(data.length, 5);
+  // test len less than 0
+  deno.truncateSync(filename, -5);
+  data = readDataSync(filename);
+  assertEqual(data.length, 0);
+  
+  // delete test txt
+  deno.removeSync(filename);
```

Comment:
I would remove all the comments here - it's self explanatory. Also remove blank links where appropriate.